dradis/dradis-ce
# Improve cache keys in commentable#show pages

### Spec
The resources `comments`, `subscriptions` and the `current_user` can be removed from the cache key in `commentable#show` view

### Proposed solution

1. Separate comments and subscriptions feed to its own cache fragment.
2. Remove current user from cache keys and use js to make specific user content visible.

### How to test

- enable cache: `rake dev:cache`

- visit a note/evidence/issue with user A
- visit the same page with user B, check that the page renders from cache
   (you may insert `<%= Time.now %>` here: https://github.com/dradis/dradis-ce/pull/384/files#diff-9c0beaca63c1fc3597ad03d826b4f02eR30)

- change subscription status with user A. Reload with user B: user B must keep the subscriptions status, and page must still be cached.

- add a comment with user A
- reload page with user B. Check that comments for user B are loaded from cache
  (you may insert `<%= Time.now %>` here: https://github.com/dradis/dradis-ce/pull/384/files#diff-1aae7ed6c016b3e62cf25303b9a3f452R4)

- Add a comment for user B. Assert that each user can only see the edit/delete links for the comments that he/she posted.
